6/11/2023 10:30:57 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '1013' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

6/11/2023 10:30:57 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '5020' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

6/11/2023 12:00:28 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '1013' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

6/11/2023 12:00:29 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '5020' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

6/11/2023 12:04:37 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '1013' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

6/11/2023 12:04:37 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '5020' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

6/11/2023 12:06:15 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '1013' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

6/11/2023 12:06:15 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '5020' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

6/11/2023 12:08:46 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '1013' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

6/11/2023 12:08:46 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Duplicate entry '5020' for key 'PRIMARY'
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

7/11/2023 12:24:51 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

7/11/2023 12:24:53 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

7/11/2023 12:24:55 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

7/11/2023 12:24:55 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

7/11/2023 12:25:34 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:25:37 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:25:53 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

7/11/2023 12:25:53 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

7/11/2023 12:25:55 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

7/11/2023 12:25:55 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

7/11/2023 12:26:21 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioSinTarjeta' to type 'Entidades.UsuarioExtranjero'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 43

7/11/2023 12:31:51 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

7/11/2023 12:31:51 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

7/11/2023 12:31:53 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

7/11/2023 12:31:53 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

7/11/2023 12:32:04 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:32:10 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:32:27 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:32:41 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:33:05 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

7/11/2023 12:33:05 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

7/11/2023 12:33:07 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

7/11/2023 12:33:07 - Error en AccesoMySql.InsertarElementosSQL: Error al insertar elementos en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.InsertarElementosSQL(List`1 elementos, String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 50

7/11/2023 12:33:34 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:36:05 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:36:20 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:38:30 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

7/11/2023 12:38:42 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error al obtener usuario por DNI y tarjeta
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioArgentino' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 48

8/11/2023 09:31:31 - Error en ArchivoMensaje.GuardarMensajesBajaEnArchivo: Error al guardar el mensaje
System.NotSupportedException: The type System.Collections.Generic.Dictionary`2[[System.Int32, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[Entidades.ArchivoMensaje+BajaData, Entidades, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]] is not supported because it implements IDictionary.
   at System.Xml.Serialization.TypeScope.GetDefaultIndexer(Type type, String memberInfo)
   at System.Xml.Serialization.TypeScope.ImportTypeDesc(Type type, MemberInfo memberInfo, Boolean directReference)
   at System.Xml.Serialization.TypeScope.GetTypeDesc(Type type, MemberInfo source, Boolean directReference, Boolean throwOnError)
   at System.Xml.Serialization.TypeScope.ImportTypeDesc(Type type, MemberInfo memberInfo, Boolean directReference)
   at System.Xml.Serialization.TypeScope.GetTypeDesc(Type type, MemberInfo source, Boolean directReference, Boolean throwOnError)
   at System.Xml.Serialization.ModelScope.GetTypeModel(Type type, Boolean directReference)
   at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping(Type type, XmlRootAttribute root, String defaultNamespace)
   at System.Xml.Serialization.XmlSerializer..ctor(Type type, String defaultNamespace)
   at System.Xml.Serialization.XmlSerializer..ctor(Type type)
   at Entidades.ArchivoMensaje.GuardarMensajesBajaEnArchivo(List`1 listaBajas) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ArchivoMensaje.cs:line 78

8/11/2023 09:32:32 - Error en ArchivoMensaje.GuardarMensajesBajaEnArchivo: Error al guardar el mensaje
System.NotSupportedException: The type System.Collections.Generic.Dictionary`2[[System.Int32, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[Entidades.ArchivoMensaje+BajaData, Entidades, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]] is not supported because it implements IDictionary.
   at System.Xml.Serialization.TypeScope.GetDefaultIndexer(Type type, String memberInfo)
   at System.Xml.Serialization.TypeScope.ImportTypeDesc(Type type, MemberInfo memberInfo, Boolean directReference)
   at System.Xml.Serialization.TypeScope.GetTypeDesc(Type type, MemberInfo source, Boolean directReference, Boolean throwOnError)
   at System.Xml.Serialization.TypeScope.ImportTypeDesc(Type type, MemberInfo memberInfo, Boolean directReference)
   at System.Xml.Serialization.TypeScope.GetTypeDesc(Type type, MemberInfo source, Boolean directReference, Boolean throwOnError)
   at System.Xml.Serialization.ModelScope.GetTypeModel(Type type, Boolean directReference)
   at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping(Type type, XmlRootAttribute root, String defaultNamespace)
   at System.Xml.Serialization.XmlSerializer..ctor(Type type, String defaultNamespace)
   at System.Xml.Serialization.XmlSerializer..ctor(Type type)
   at Entidades.ArchivoMensaje.GuardarMensajesBajaEnArchivo(List`1 listaBajas) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ArchivoMensaje.cs:line 78

8/11/2023 10:10:44 - Error en ArchivoMensaje.GuardarMensajesBajaEnArchivo: Error al guardar el mensaje
System.NotSupportedException: The type System.Collections.Generic.Dictionary`2[[System.Int32, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[Entidades.ArchivoMensaje+BajaData, Entidades, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]] is not supported because it implements IDictionary.
   at System.Xml.Serialization.TypeScope.GetDefaultIndexer(Type type, String memberInfo)
   at System.Xml.Serialization.TypeScope.ImportTypeDesc(Type type, MemberInfo memberInfo, Boolean directReference)
   at System.Xml.Serialization.TypeScope.GetTypeDesc(Type type, MemberInfo source, Boolean directReference, Boolean throwOnError)
   at System.Xml.Serialization.TypeScope.ImportTypeDesc(Type type, MemberInfo memberInfo, Boolean directReference)
   at System.Xml.Serialization.TypeScope.GetTypeDesc(Type type, MemberInfo source, Boolean directReference, Boolean throwOnError)
   at System.Xml.Serialization.ModelScope.GetTypeModel(Type type, Boolean directReference)
   at System.Xml.Serialization.XmlReflectionImporter.ImportTypeMapping(Type type, XmlRootAttribute root, String defaultNamespace)
   at System.Xml.Serialization.XmlSerializer..ctor(Type type, String defaultNamespace)
   at System.Xml.Serialization.XmlSerializer..ctor(Type type)
   at Entidades.ArchivoMensaje.GuardarMensajesBajaEnArchivo(List`1 listaBajas) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ArchivoMensaje.cs:line 77

9/11/2023 09:50:35 - Error en UsuarioArgentino.InsertarElementoSQL: Error al insertar elemento a la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Table 'proyectosube.usuarioextrangero' doesn't exist
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.UsuarioExtranjero.InsertarElementoSQL(UsuarioExtranjero usuariosExtrangero) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\UsuarioExtranjero.cs:line 94

9/11/2023 10:32:08 - Error en UsuarioExtranjero.InsertarElementoSQL: Error al insertar elemento a la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Table 'proyectosube.usuarioextrangero' doesn't exist
   at MySql.Data.MySqlClient.MySqlStream.ReadPacketAsync(Boolean execAsync)
   at MySql.Data.MySqlClient.NativeDriver.GetResultAsync(Int32 affectedRow, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.GetResultAsync(Int32 statementId, Int32 affectedRows, Int64 insertedId, Boolean execAsync)
   at MySql.Data.MySqlClient.Driver.NextResultAsync(Int32 statementId, Boolean force, Boolean execAsync)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlDataReader.NextResultAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.UsuarioExtranjero.InsertarElementoSQL(UsuarioExtranjero usuariosExtrangero) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\UsuarioExtranjero.cs:line 95

10/11/2023 10:25:02 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 10:25:02 - Error en AccesoMySql.ObtenerElementosSQL: Error al obtener elementos de la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader()
   at Entidades.AccesoMySql`1.ObtenerElementosSQL(String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 84

10/11/2023 10:25:04 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 10:25:04 - Error en AccesoMySql.ObtenerElementosSQL: Error al obtener elementos de la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader()
   at Entidades.AccesoMySql`1.ObtenerElementosSQL(String nombreTabla) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 84

10/11/2023 10:25:06 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 10:25:06 - Error en UsuarioSinTarjeta.ObtenerElementosSQL: Error al obtener elementos de la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader()
   at Entidades.UsuarioSinTarjeta.ObtenerElementosSQL() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\UsuarioSinTarjeta.cs:line 45

10/11/2023 10:25:08 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 10:25:08 - Error en UsuarioArgentino.ObtenerElementosSQL: Error al obtener elementos de la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader()
   at Entidades.UsuarioArgentino.ObtenerElementosSQL() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\UsuarioArgentino.cs:line 85

10/11/2023 10:25:10 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 10:25:10 - Error en UsuarioExtranjero.ObtenerElementosSQL: Error al obtener elementos de la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader()
   at Entidades.UsuarioExtranjero.ObtenerElementosSQL() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\UsuarioExtranjero.cs:line 50

10/11/2023 10:25:12 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.
 ---> System.Net.Sockets.SocketException (10061): No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task)
   at MySql.Data.MySqlClient.NativeDriver.<>c.<OpenAsync>b__38_0(Exception ex)
   at System.AggregateException.Handle(Func`2 predicate)
   at MySql.Data.MySqlClient.NativeDriver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.Driver.CreateAsync(MySqlConnectionStringBuilder settings, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.CreateNewPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetPooledConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.TryToGetDriverAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlPool.GetConnectionAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 10:25:12 - Error en UsuarioAdministrador.ObtenerElementosSQL: Error al obtener elementos de la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReader()
   at Entidades.UsuarioAdministrador.ObtenerElementosSQL() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\UsuarioAdministrador.cs:line 34

10/11/2023 10:51:10 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error para el traer el objeto de la base de datos
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioAdministrador' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 33

10/11/2023 10:52:04 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error para el traer el objeto de la base de datos
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioAdministrador' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 33

10/11/2023 11:04:40 - Error en ManejoDeListados.ObtenerUsuarioPorDniYTarjeta: Error para el traer el objeto de la base de datos
System.InvalidCastException: Unable to cast object of type 'Entidades.UsuarioAdministrador' to type 'Entidades.UsuarioSinTarjeta'.
   at Entidades.ManejoDeListados.ObtenerUsuarioPorDniYTarjeta(String dni) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ManejoDeListados.cs:line 33

10/11/2023 18:05:06 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:05:06 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:05:07 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 32
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 143

10/11/2023 18:05:07 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 32
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 143

10/11/2023 18:05:07 - Error en TarjetaNacional.ActualizarEnBaseDeDatos: Error al actualizar tabla mySQL de tarjeta nacional
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 188
   at Entidades.TarjetaNacional.ActualizarEnBaseDeDatos(String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\TarjetaNacional.cs:line 137

10/11/2023 18:05:07 - Error en TarjetaNacional.ActualizarEnBaseDeDatos: Error al actualizar tabla mySQL de tarjeta nacional
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 188
   at Entidades.TarjetaNacional.ActualizarEnBaseDeDatos(String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\TarjetaNacional.cs:line 137

10/11/2023 18:15:33 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:15:33 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 32
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 143

10/11/2023 18:15:34 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:15:41 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:15:41 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:15:41 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:15:41 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:15:41 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:15:41 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:18:09 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:18:09 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 32
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 143

10/11/2023 18:18:09 - Error en TarjetaNacional.ActualizarEnBaseDeDatos: Error al actualizar tabla mySQL de tarjeta nacional
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 188
   at Entidades.TarjetaNacional.ActualizarEnBaseDeDatos(String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\TarjetaNacional.cs:line 137

10/11/2023 18:18:09 - Error en TarjetaNacional.ActualizarEnBaseDeDatos: Error al actualizar tabla mySQL de tarjeta nacional
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 188
   at Entidades.TarjetaNacional.ActualizarEnBaseDeDatos(String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\TarjetaNacional.cs:line 137

10/11/2023 18:18:09 - Error en TarjetaNacional.ActualizarEnBaseDeDatos: Error al actualizar tabla mySQL de tarjeta nacional
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 188
   at Entidades.TarjetaNacional.ActualizarEnBaseDeDatos(String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\TarjetaNacional.cs:line 137

10/11/2023 18:21:03 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:21:03 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 32
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 143

10/11/2023 18:21:03 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 32
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 143

10/11/2023 18:21:03 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 32
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 143

10/11/2023 18:21:03 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:21:03 - Error en TarjetaNacional.ActualizarEnBaseDeDatos: Error al actualizar tabla mySQL de tarjeta nacional
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 188
   at Entidades.TarjetaNacional.ActualizarEnBaseDeDatos(String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\TarjetaNacional.cs:line 137

10/11/2023 18:21:03 - Error en TarjetaNacional.ActualizarEnBaseDeDatos: Error al actualizar tabla mySQL de tarjeta nacional
System.IO.IOException: The process cannot access the file 'C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\ProyectoSUBEAlfonzoFatala\Archivos\error_log.txt' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.AppendAllText(String path, String contents)
   at Entidades.CatchError.LogError(String className, String methodName, String description, Exception exception) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\CatchError.cs:line 22
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 188
   at Entidades.TarjetaNacional.ActualizarEnBaseDeDatos(String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\TarjetaNacional.cs:line 137

10/11/2023 18:21:11 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:21:11 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
System.InvalidOperationException: Connection must be valid and open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:21:11 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:21:11 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:21:11 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:21:11 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:21:11 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:21:11 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:21:11 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:21:11 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:21:11 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:21:11 - Error en ConexionSQL.Conectar: Error al conectar la base de datos
System.InvalidOperationException: The connection is already open.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at Entidades.ConexionSQL.Conectar() in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\ConexionSQL.cs:line 28

10/11/2023 18:21:11 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

10/11/2023 18:21:11 - Error en AccesoMySql.ActualizarElementoSQL: Error al actualizar elemento en la base de datos
MySql.Data.MySqlClient.MySqlException (0x80004005): There is already an open DataReader associated with this Connection which must be closed first.
   at MySql.Data.MySqlClient.Interceptors.ExceptionInterceptor.Throw(Exception exception)
   at MySql.Data.MySqlClient.MySqlCommand.CheckState()
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteReaderAsync(CommandBehavior behavior, Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(Boolean execAsync, CancellationToken cancellationToken)
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery()
   at Entidades.AccesoMySql`1.ActualizarElementoSQL(T elemento, String nombreTabla, String condicion) in C:\Users\usuario\source\repos\ProyectoSUBEAlfonzoFatala\Entidades\AccesoMySql.cs:line 182

